solutions/usaco-834.mdx

```diff
@@ -0,0 +1,71 @@
+---
+id: usaco-1037
+source: USACO Silver 2018 US Open
+title: Out of Sorts
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_sort_silver_open18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class OutOfSorts {
+
+	// hold entry values
+	static class Entry implements Comparable<Entry> {
+		public int value;
+		public int index;
+
+		public Entry (int v, int i) {
+			value = v;
+			index = i;
+		}
+
+		public int compareTo(Entry other) { 
+			if (this.value != other.value) {
+				return this.value - other.value;
+			}
+			return this.index - other.index;
+		}
+	}
+
+	public static void main(String[] args) throws Exception {
+		Kattio io = new Kattio("sort");
+		int n = io.nextInt();
+		Entry[] entries = new Entry[n];
+		for (int i = 0; i < n; i++) {
+			entries[i] = new Entry(io.nextInt(), i);
+		}
+				      
+		// sort the values so that the value of each array element indices 
+		// the index where that element should go in the sorted array
+		Arrays.sort(entries); 
+
+		// values usually only move one spot per iteration
```

Comment:
not clear what this means in the context of ans being initialized to 1

Reply:
in bubble sort elements only move left one. i think I will remove because its not relevant